### DISCUSSION TITLE
ConversationReferenceStoreAddOptions - Why is Overwrite needed?

### DISCUSSION THREAD

**Discussion by arjavgarg** [OfficeDev/microsoft-365-agents-toolkit#11030] [Q&A]
Hello, I was wondering what the purpose of `Overwrite` flag in `ConversationReferenceStoreAddOptions` in `Add` method of `IConversationReferenceStore` is. Shouldn't we always just overwrite with the latest value, regardless of whether the `Activity` was about a user installing the Bot or messaging the Bot?

@qinezh @swatDong @dooriya 

https://github.com/OfficeDev/TeamsFx/pull/4981 https://github.com/OfficeDev/TeamsFx/pull/8486

Thanks

**Comment by qinezh** [comment]
It would cause performance issues if conversation reference were always written into database when users message the bot.

**Reply by arjavgarg** [reply]
@qinezh However, create query has to be run every time right? It is just the difference between an update vs an insert query. Query has to be made in both the cases.

Don't you agree?
